controllers/challenge_controller.go

```diff
@@ -0,0 +1,100 @@
+package controllers
+
+import (
+	"net/http"
+
+	"github.com/drift-org/backend/helpers"
+	"github.com/drift-org/backend/models"
+	"github.com/gin-gonic/gin"
+	"github.com/kamva/mgm/v3"
+	"go.mongodb.org/mongo-driver/bson"
+)
+
+type ChallengeController interface {
+	Create(context *gin.Context)
+}
+
+type challengeController struct{}
+
+func NewChallengeController() ChallengeController {
+	return &challengeController{}
+}
+
+// TODO: Idea for future, if we want to let companies/orgs create challenges (Business Model #3).
+// Default method of adding challenges on frontend is to add by placing a marker on a map
+// (allowing us to grab Lat/Long coordinates). We can also offer a (premium) option for users to
+// provide just the address, and convert to Lat/Long -- thus allowing batch creation of challenges.
+func (ctrl *challengeController) Create(context *gin.Context) {
+	type ICreate struct {
+		Latitude    float64 `json:"latitude"`
+		Longitude   float64 `json:"longitude"`
+		Address     string  `json:"address"`
+		TaskName    string  `json:"taskName" binding:"required"`
+		Description string  `json:"description" binding:"required"`
+		Points      int     `json:"points" binding:"required,min=0"`
+	}
+	var body ICreate
+	if err := context.ShouldBindJSON(&body); err != nil {
+		context.AbortWithStatusJSON(http.StatusBadRequest, gin.H{"error": err.Error()})
+		return
+	}
+	location := models.Location{}
+
+	challenge := models.Challenge{
+		TaskName:    body.TaskName,
+		Description: body.Description,
+		Points:      body.Points,
+	}
+	location.Type = "Point"
+
+	// If address information is provided, include it in our Challenge model.
+	if body.Address != "" {
```

Comment:
We should have a discussion about what sort of work is done inside the route and what sort of work is done inside the `Creating` method inside the model.

I haven't thought about pros/cons, but i could see how one could argue that this logic could belong in the `Creating` of the model, especially since the cases are documented in that file and it would be nice to point to what cases correspond to what pieces of code.

Reply:
Is this referring to setting the address/lat/long part? Right now only thing is that challenge model contains information in a different format (refers to the Location model). If we want we can move it there later, but for now I feel like this is fine?